app/src/main/java/com/coveros/coverosmobileapp/blogpost/BlogPostsListActivity.java

```diff
@@ -163,16 +154,16 @@ public void onSuccess(List<BlogPost> newPosts) {
     protected void retrieveAuthors(final PostListCallback<String> postListCallback) {
         StringRequest authorsRequest = new StringRequest(Request.Method.GET, AUTHORS_URL, new Response.Listener<String>() {
             @Override
-            public void onResponse(String response) {
-                JsonArray authorsJson = new JsonParser().parse(response).getAsJsonArray();
-                for (JsonElement author : authorsJson) {
-                    JsonObject authorJson = (JsonObject) author;
-                    Integer id = authorJson.get("id").getAsInt();
-                    authors.put(id, authorJson.get("name").getAsString());
-                }
-                postListCallback.onSuccess(null);
+        public void onResponse(String response) {
+            JsonArray authorsJson = new JsonParser().parse(response).getAsJsonArray();
+            for (JsonElement author : authorsJson) {
+                JsonObject authorJson = (JsonObject) author;
```

Comment:
See other comment. This is code that is repeated and should be encapsulated somewhere else so we don't violate the DRY principla